modules/ROOT/pages/directives/indexes-and-constraints.adoc

```diff
@@ -247,3 +247,167 @@ const schema = await neoSchema.getSchema();
 
 await neoSchema.assertIndexesAndConstraints({ options: { create: true }});
 ----
+
+
+:description: Directives related to generative AI in the Neo4j GraphQL Library.
+
+
+== Vector index search
+
+With the `@vector` GraphQL directive you can query your database to perform a vector index search.
+Queries are performed by passing in either a vector index or a query phrase.
+
+A query by vector index finds nodes with a vector embedding similar to that index.
+That is, the query performs a nearest neighbor search.
+
+In contrast, a query by phrase (a string of text) forwards the phrase to the link:https://neo4j.com/docs/cypher-manual/current/genai-integrations/[Neo4j GenAI plugin] and the plugin generates a vector embedding for it.
+This embedding is then compared to the node vector embeddings in the database.
+
+[NOTE] 
+.Prerequisites
+==== 
+* The database must be Neo4j version 5.15 or higher.
+* The node vector embeddings already exist in the database.
+* The embeddings must have been created using the same method.
+* Queries by vector index cannot be performed across multiple labels.
+* Queries by phrase require credentials for the Neo4j GenAI plugin.
+====
+
+[NOTE]
+====
+Vector index searches are _read-only_ in the sense that no data is persisted in the database.
```

Comment:
When I read "no data is persisted in the database" it almost conjures the thought that it will delete the existing data. I know that's incorrect, but perhaps we can change the wording of this?

Reply:
changed the wording